python/triton/language/libdevice.py

```diff
@@ -0,0 +1,1353 @@
+from . import core, extern
+
+LIBDEVICE_PATH = "/usr/local/cuda/nvvm/libdevice/libdevice.10.bc"
```

Comment:
the path is hardcoded here? Will this file be regenerated if the user passes a different path?

Reply:
Users can pass libdevice pass on a per-kernel basis. But your comment makes me think that we do need indeed to give users the ability to provide an environment variable name that overrides this default when creating the file. We were just talking about it yesterday with @Jokeren  :)

Reply:
Another idea would be to bundle it with triton at build time.

Reply:
Triton's build does not depend on any CUDA path any more. 

Our idea is allowing users to pass through the path through `os.getenv("TRITON_LIBDEVICE_PATH")`. What do you think?

Reply:
One thing that would help would be to make `libdevice.py` path-independent. Then, as done currently, users run the kernel with:
```
kernel[grid](..., libs={'libdevice': path..., 'some_other_lib'=other_path})
```
with the value of `libs` defaulting to:
```
libs = {'libdevice': /path/to/ptxas/../nvvm/libdevice/libdevice.10.bc'}.
```
where /path/to/ptxas uses this function https://github.com/openai/triton/blob/2bed6fc850b93854d7f411e77c9a302954604091/lib/driver/llvm.cc#L96

Reply:
@Jokeren  Let's package `libdevice.10.bc`. My understanding of the EULA https://docs.nvidia.com/cuda/eula/index.html#cuda-toolkit-supplement-license-agreement is that it is allowed.

> The following CUDA Toolkit files may be distributed with Licensee Applications developed by you, including certain variations of these files that have version number or architecture specific information embedded in the file name - as an example only, for release version 9.0 of the 64-bit Windows software, the file cudart64_90.dll is redistributable.
libdevice.10.bc

Reply:
Do you mean copy `libdevice.10.bc` to a path like `triton/language`, then the default libdevice path would be `./libdevice.10.bc`?

Reply:
Fixed in the new commit